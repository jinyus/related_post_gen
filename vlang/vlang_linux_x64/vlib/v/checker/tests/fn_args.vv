fn uu8(a u8) {}

fn arr(a []int) {}

fn fun(a fn (int)) {}

fn basic() {
	v := 4
	uu8(&v)
	arr([5]!)
	fun(fn (i &int) {})
	fun(fn (ii ...int) {})
}

struct S1 {
}

fn f(p &S1) {}

fn ptr() {
	v := 4
	f(v)
}
