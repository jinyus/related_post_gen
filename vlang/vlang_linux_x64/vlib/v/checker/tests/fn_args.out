vlib/v/checker/tests/fn_args.vv:9:6: error: cannot use `&int` as `u8` in argument 1 to `uu8`
    7 | fn basic() {
    8 |     v := 4
    9 |     uu8(&v)
      |         ~~
   10 |     arr([5]!)
   11 |     fun(fn (i &int) {})
vlib/v/checker/tests/fn_args.vv:10:6: error: cannot use `[1]int` as `[]int` in argument 1 to `arr`
    8 |     v := 4
    9 |     uu8(&v)
   10 |     arr([5]!)
      |         ~~~~
   11 |     fun(fn (i &int) {})
   12 |     fun(fn (ii ...int) {})
vlib/v/checker/tests/fn_args.vv:11:6: error: cannot use `fn (&int)` as `fn (int)` in argument 1 to `fun`
    9 |     uu8(&v)
   10 |     arr([5]!)
   11 |     fun(fn (i &int) {})
      |         ~~~~~~~~~~~~~~
   12 |     fun(fn (ii ...int) {})
   13 | }
Details: expected argument 1 to be NOT a pointer, but the passed argument 1 is a pointer
vlib/v/checker/tests/fn_args.vv:12:6: error: cannot use `fn (...int)` as `fn (int)` in argument 1 to `fun`
   10 |     arr([5]!)
   11 |     fun(fn (i &int) {})
   12 |     fun(fn (ii ...int) {})
      |         ~~~~~~~~~~~~~~~~~
   13 | }
   14 |
vlib/v/checker/tests/fn_args.vv:22:4: error: cannot use `int` as `&S1` in argument 1 to `f`
   20 | fn ptr() {
   21 |     v := 4
   22 |     f(v)
      |       ^
   23 | }
